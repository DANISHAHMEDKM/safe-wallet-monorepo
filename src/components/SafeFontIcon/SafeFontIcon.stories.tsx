import { View, Text, ScrollView } from 'react-native'
import type { Meta, StoryObj } from '@storybook/react'
import { SafeFontIcon } from './SafeFontIcon'
import { iconNames } from '@/src/types/iconTypes'

// Meta information for the Story
const meta: Meta<typeof SafeFontIcon> = {
  // title: 'Icons/AllIcons',
  component: SafeFontIcon,
  argTypes: {
    color: { control: 'color' },
  },
}

export default meta

// Create a type alias for the Story
type Story = StoryObj<typeof SafeFontIcon>

// Define the Story for displaying all icons
export const AllIcons: Story = {
  render: (args) => (
    <ScrollView contentContainerStyle={{ padding: 10 }}>
      <View style={{ flexDirection: 'row', flexWrap: 'wrap', justifyContent: 'space-between' }}>
        {iconNames.map((iconName) => (
          <View key={iconName} style={{ width: '30%', alignItems: 'center', marginBottom: 20 }}>
            <SafeFontIcon {...args} name={iconName} />
            <Text style={{ marginTop: 10, fontWeight: 'bold' }}>{iconName}</Text>
          </View>
        ))}
      </View>
    </ScrollView>
  ),
  args: {
    size: 50,
    color: 'black',
  },
}
