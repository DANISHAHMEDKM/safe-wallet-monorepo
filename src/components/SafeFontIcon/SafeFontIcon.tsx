import React from 'react'
import createIconSetFromIcoMoon from '@expo/vector-icons/createIconSetFromIcoMoon'
import { useFonts } from 'expo-font'
import { IconName } from '@/src/types/iconTypes'

export const SafeIcon = createIconSetFromIcoMoon(
  require('@/assets/fonts/safe-icons/selection.json'),
  'SafeIcons',
  'safe-icons.ttf',
)

export interface IconProps {
  name: IconName
  size?: number
  color?: string
}

export const SafeFontIcon = ({ name, size = 24, color, ...rest }: IconProps) => {
  const [fontsLoaded] = useFonts({
    SafeIcons: require('@/assets/fonts/safe-icons/safe-icons.ttf'),
  })

  if (!fontsLoaded) {
    return null
  }

  return <SafeIcon name={name} size={size} color={color} {...rest} />
}
