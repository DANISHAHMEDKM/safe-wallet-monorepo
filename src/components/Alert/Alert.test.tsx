import { render } from '@/src/tests/test-utils'
import { Alert } from '.'
import { SafeFontIcon } from '../SafeFontIcon/SafeFontIcon'

describe('Alert', () => {
  it('should render a info alert', () => {
    const container = render(<Alert type="info" message="Info alert" />)

    expect(container).toMatchSnapshot()
  })

  it('should render a warning alert', () => {
    const container = render(<Alert type="warning" message="Warning alert" />)

    expect(container).toMatchSnapshot()
  })

  it('should render an error alert', () => {
    const container = render(<Alert type="error" message="Error alert" />)

    expect(container).toMatchSnapshot()
  })

  it('should render an alert with start icon', () => {
    const container = render(<Alert type="error" startIcon={<SafeFontIcon name="add-owner" />} message="Error alert" />)

    expect(container).toMatchSnapshot()
  })

  it('should render an alert with an end icon', () => {
    const container = render(<Alert type="error" endIcon={<SafeFontIcon name="add-owner" />} message="Error alert" />)

    expect(container).toMatchSnapshot()
  })

  it('should render an alert with a name icon', () => {
    const container = render(<Alert type="error" iconName="add-owner" message="Error alert" />)

    expect(container).toMatchSnapshot()
  })
})
