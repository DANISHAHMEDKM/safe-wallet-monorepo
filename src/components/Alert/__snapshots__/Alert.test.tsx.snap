// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render a info alert 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={false}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#EFFCFF",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flexDirection": "row",
            "gap": 12,
            "justifyContent": "center",
            "paddingBottom": 12,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 12,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#5FDDFF",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "SafeIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "color": "#5FDDFF",
              "fontFamily": "DMSans-SemiBold",
              "fontSize": 16,
            }
          }
          suppressHighlighting={true}
        >
          Info alert
        </Text>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`Alert should render a warning alert 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={false}
  exclusive={true}
  handlerTag={2}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff0e0",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flexDirection": "row",
            "gap": 12,
            "justifyContent": "center",
            "paddingBottom": 12,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 12,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#FF8C00",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "SafeIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "color": "#FF8C00",
              "fontFamily": "DMSans-SemiBold",
              "fontSize": 16,
            }
          }
          suppressHighlighting={true}
        >
          Warning alert
        </Text>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`Alert should render an alert with a name icon 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={false}
  exclusive={true}
  handlerTag={6}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#49191F",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flexDirection": "row",
            "gap": 12,
            "justifyContent": "center",
            "paddingBottom": 12,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 12,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#FF5F72",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "SafeIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "color": "#FF5F72",
              "fontFamily": "DMSans-SemiBold",
              "fontSize": 16,
            }
          }
          suppressHighlighting={true}
        >
          Error alert
        </Text>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`Alert should render an alert with an end icon 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={false}
  exclusive={true}
  handlerTag={5}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#49191F",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flexDirection": "row",
            "gap": 12,
            "justifyContent": "center",
            "paddingBottom": 12,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 12,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#FF5F72",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "SafeIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "color": "#FF5F72",
              "fontFamily": "DMSans-SemiBold",
              "fontSize": 16,
            }
          }
          suppressHighlighting={true}
        >
          Error alert
        </Text>
        <View>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#FF5F72",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "SafeIcons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`Alert should render an alert with start icon 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={false}
  exclusive={true}
  handlerTag={4}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#49191F",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flexDirection": "row",
            "gap": 12,
            "justifyContent": "center",
            "paddingBottom": 12,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 12,
            "width": "100%",
          }
        }
      >
        <View>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#FF5F72",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "SafeIcons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#FF5F72",
              "fontFamily": "DMSans-SemiBold",
              "fontSize": 16,
            }
          }
          suppressHighlighting={true}
        >
          Error alert
        </Text>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`Alert should render an error alert 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={false}
  exclusive={true}
  handlerTag={3}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#49191F",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flexDirection": "row",
            "gap": 12,
            "justifyContent": "center",
            "paddingBottom": 12,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 12,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#FF5F72",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "SafeIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "color": "#FF5F72",
              "fontFamily": "DMSans-SemiBold",
              "fontSize": 16,
            }
          }
          suppressHighlighting={true}
        >
          Error alert
        </Text>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;
